main/test/ca/uwaterloo/flix/library/TestEnvironment.flix

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2021 Ramiro Calle
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace EnvironmentTest {
+
+  @test
+  def testGetEnv01(): Bool =
+    Map.isEmpty(Environment.getEnv()) == false
+
+  @test
+  def testGetEnv02(): Bool =
+    Environment.getEnv() |> Map.get("PATH") |> Option.isEmpty == false
+
+  @test
+  def testGetEnv03(): Bool & Impure = {
+      import new java.io.FileInputStream(String) as newFileInputStream;
+      import new java.util.Properties() as newProperties;
+      import java.util.Properties.getProperty(String);
+      import java.util.Properties.load(##java.io.InputStream);
+      import java.util.Properties.propertyNames();
+      import java.util.Enumeration.hasMoreElements();
+      import java.util.Enumeration.nextElement();
+      use List.flatMap;
+      let* filename = Option.toList(Environment.getProp("TEST_ENV_FILE"));
+      let in = newFileInputStream(filename) as ##java.io.InputStream;
+      let prop = newProperties();
+      prop.load(in);
+      let names = prop.propertyNames();
+      let* expectedKey = Iterator.toList(Iterator(() -> not names.hasMoreElements(), () -> (names.nextElement() as String)));
+      let expectedValue = prop.getProperty(expectedKey) as String;
+      let actualValues = Environment.getEnv();
+      (Map.get(expectedKey, actualValues) |> Option.exists(actualValue -> actualValue == expectedValue)) :: Nil
+    } |> List.forall(x -> x)
+
+  @test
+  def testGetVar01(): Bool =
+    Option.isEmpty(Environment.getVar("PATH")) == false
+
+  @test
+  def testGetVar02(): Bool =
+    Option.isEmpty(Environment.getVar("UNKNOWN"))
+
+  @test
+  def testgetProp01(): Bool =
+    Environment.getProp("TEST_PROPERTY") == Some("Hello world!")
+
+  @test
+  def testgetProp02(): Bool =
+    Option.isEmpty(Environment.getProp("UNKNOWN"))
+
+  @test
+  def testgetOsName(): Bool =
+    Option.isEmpty(Environment.getOsName()) == false
+
+  @test
+  def testgetOsArch(): Bool =
+    Option.isEmpty(Environment.getOsArch()) == false
+
+  @test
+  def testgetOsVersion(): Bool =
+    Option.isEmpty(Environment.getOsVersion()) == false
```

Comment:
not Option.isEmpty(Environment.getOsVersion())

Reply:
same for the rest.

Reply:
ok